ProcessMaker/Events/ScreenUpdated.php

```diff
@@ -28,6 +29,13 @@ public function __construct(Screen $screen, array $changes, array $original)
         $this->screen = $screen;
         $this->changes = $changes;
         $this->original = $original;
+
+        if (isset($original['screen_category_id']) && isset($changes['screen_category_id'])) {
```

Comment:
I understand the goal of the code suggested. However, if we send "int" value as parameter, the unset will reutrn error. The function getNamesByIds receive String values, not objects. Implement a function will add more code lines to the event and make more complex.

Reply:
@CarliPinell as I can see the current dependency from the original and changes are correct because is possible to have values in $original and not in $changes, in this way please make the following changes:

```
// Get category name
        $this->original['screen_category'] = isset($original['screen_category_id']) ? ScreenCategory::getNamesByIds($this->original['screen_category_id']) : '';
        unset($this->original['screen_category_id']);
        $this->changes['screen_category'] = isset($changes['screen_category_id']) ? ScreenCategory::getNamesByIds($this->changes['screen_category_id']) : '';
        unset($this->changes['screen_category_id']);
```

Reply:
Understood. Solved